keyboards/model_f_labs/f62/wcass/info.json

```diff
@@ -0,0 +1,634 @@
+                {"label": "I", "matrix": [5, 4], "x": 8.5, "y": 1},
+                {"label": "O", "matrix": [7, 5], "x": 9.5, "y": 1},
+                {"label": "P", "matrix": [5, 5], "x": 10.5, "y": 1},
+                {"label": "{", "matrix": [5, 6], "x": 11.5, "y": 1},
+                {"label": "}", "matrix": [5, 7], "x": 12.5, "y": 1},
+                {"label": "Enter", "matrix": [2, 7], "x": 13.75, "y": 1, "w": 1.25, "h": 2},
+                {"label": "Ctrl", "matrix": [7, 0], "x": 0, "y": 2, "w": 1.75},
+                {"label": "A", "matrix": [7, 1], "x": 1.75, "y": 2},
+                {"label": "S", "matrix": [2, 1], "x": 2.75, "y": 2},
+                {"label": "D", "matrix": [0, 2], "x": 3.75, "y": 2},
+                {"label": "F", "matrix": [2, 2], "x": 4.75, "y": 2},
+                {"label": "G", "matrix": [0, 3], "x": 5.75, "y": 2},
+                {"label": "H", "matrix": [2, 3], "x": 6.75, "y": 2},
+                {"label": "J", "matrix": [2, 4], "x": 7.75, "y": 2},
+                {"label": "K", "matrix": [0, 4], "x": 8.75, "y": 2},
+                {"label": "L", "matrix": [0, 5], "x": 9.75, "y": 2},
+                {"label": ":", "matrix": [2, 5], "x": 10.75, "y": 2},
+                {"label": "\"", "matrix": [2, 6], "x": 11.75, "y": 2},
+                {"label": "nuhs", "matrix": [7, 6], "x": 12.75, "y": 2},
+                {"label": "Shift", "matrix": [2, 0], "x": 0, "y": 3, "w": 1.25},
+                {"label": "nubs", "matrix": [0, 0], "x": 1.25, "y": 3},
+                {"label": "Z", "matrix": [1, 1], "x": 2.25, "y": 3},
+                {"label": "X", "matrix": [0, 1], "x": 3.25, "y": 3},
+                {"label": "C", "matrix": [1, 2], "x": 4.25, "y": 3},
+                {"label": "V", "matrix": [3, 2], "x": 5.25, "y": 3},
+                {"label": "B", "matrix": [1, 3], "x": 6.25, "y": 3},
+                {"label": "N", "matrix": [1, 4], "x": 7.25, "y": 3},
+                {"label": "M", "matrix": [3, 4], "x": 8.25, "y": 3},
+                {"label": "<", "matrix": [3, 5], "x": 9.25, "y": 3},
+                {"label": ">", "matrix": [1, 5], "x": 10.25, "y": 3},
+                {"label": "?", "matrix": [0, 6], "x": 11.25, "y": 3},
+                {"label": "Shift", "matrix": [0, 7], "x": 12.25, "y": 3, "w": 1.75},
+                {"label": "Fn", "matrix": [1, 7], "x": 14, "y": 3},
+                {"label": "Caps Lock", "matrix": [3, 0], "x": 0, "y": 4, "w": 1.5},
+                {"label": "K41", "matrix": [1, 0], "x": 1.5, "y": 4},
+                {"label": "Alt", "matrix": [3, 1], "x": 2.5, "y": 4, "w": 1.5},
+                {"label": "K43", "matrix": [3, 3], "x": 4, "y": 4, "w": 7},
+                {"label": "Alt", "matrix": [1, 6], "x": 11, "y": 4, "w": 1.5},
+                {"label": "Num Lock", "matrix": [3, 7], "x": 12.5, "y": 4},
+                {"label": "Ctrl", "matrix": [3, 8], "x": 13.5, "y": 4, "w": 1.5}
+            ]
+        }
+    }
+}
```

Comment:
The Model F5 Labs keyboards do not have the community layouts. They have a unique layout.

Reply:
Drive-by comment from a random QMK user who recently got an F62 and has been watching this PR: The F62 PCB definitely supports some community layouts. For example, mine is [set up as `60_tsangan_hhkb`](https://media.discordapp.net/attachments/1067521540784132299/1190449142871367720/PXL_20231230_001656917.jpg).

Reply:
@matthew-wolf-n4mtt 
Looking at [image gallery](https://imgur.com/a/gw8FOl0), I can see there's also ABNT2/JIS split Right Shift (1u-1.75u) available on this board.
Would it be correct that matrix co-ordinates for 1u and 1.75u are `0,7` and `1,7` respectively?

Reply:
Not sure of the distinction between "unique" and "community", exactly, but here is my 2c as well @matthew-wolf-n4mtt:

On my F62, I've added flippers on every possible contact point.

See [here](https://postimg.cc/QVNwVCRX/180f46e2) and [here](https://i.postimg.cc/bwkkdg9w/2024-01-02-17-12.png)

Reply:
There is community interest in adding the community layouts. I will work on add them. It's going to take me a good amount of time to add and test the F62 and F77 community layouts.  It's going to delay the pull request!

The F62 and F77 is reproduction of the keyboards for the IBM 4704 which was released in 1982.  There may not be a direct mapping to all of the community layouts.  

There are key maps for the most common key use. It was requested to limit the number of files in pull request. This pull request only has two key maps (default and via) in order to limit the number files that have to be reviewed.

There is a total of 8 key maps for the F62. The other key maps will be in a follow on pull request.
You can see the all the key maps in my development branch.
[model_f_labs_f62_f77-pull](https://github.com/matthew-wolf-n4mtt/qmk_firmware/tree/model_f_labs_f62_f77-pull/keyboards/model_f_labs)

Reply:
The key maps that model f labs provided use the "LAYOUT_all" layout. There are place holder codes for the key caps and flippers that are not installed that are not installed. 

There is currently 30 key maps for the F77. There are going to be a large number of pull requests for the F77.

Reply:
Sorry if I'm just being dense, but does supporting community layouts in fact require adding new files to the pull request? I think it's all done through the keyboard's JSON config now. [See example from an existing 60% keyboard](https://github.com/qmk/qmk_firmware/blob/b02abcb9a63cc64d050f5d714ab6f2c0a4e278e6/keyboards/ai03/polaris/info.json#L45-L331).

But I'm not aware of any requirement to add a keymap for every layout a keymap supports (and I'm not seeing such a requirement in the [PR checklist](https://docs.qmk.fm/#/pr_checklist) either). All that's required is ensuring the matrix positions for each community layout are correct.

Reply:
Excerpt from PR Checklist:
- If the keyboard has multiple electrical/switch layouts:
    - include a LAYOUT_all which specifies all possible layout positions in the electrical matrix
    - use alternate layout names for all other possible layouts, preferring community layout names if an equivalent is available (e.g. LAYOUT_tkl_ansi, LAYOUT_ortho_4x4 etc.)

@bcat is correct in that there isn't a requirement to have a keynap for every layout defined in `info.json`.

[Community Layout suggestion](https://github.com/qmk/qmk_firmware/pull/21193#discussion_r1436858416) adheres to PR Checklist.

Reply:
I got confused by the physical layout of the F77.   Which is tsangan plus macro keys.

The physical layout of the F62 is tsangan with an additional key under the right side of the space bar. 

IBM 4704 -> tsangan ????? 

What dunk2k proposed is correct. I only realized this after I went did everything dunk2k all ready had done. 

Can the '60_tsangan_hhkb" layout be changed to "60_ansi_tsangan_split_bs_rshift" to match the naming of the other tsangan community layouts.

The affect of change to the community layout naming.

LAYOUT_ansi_regular_shift_regular_backspace -> LAYOUT_60_ansi_tsangan
LAYOUT_ansi_regular_shift_split_backspace -> LAYOUT_60_iso_tsangan_split_bs       (Not Community Layout)
LAYOUT_ansi_split_shift_regular_backspace -> LAYOUT_60_ansi_tsangan_split_rshift (Not Community Layout)
LAYOUT_ansi_split_shift_split_backspace -> LAYOUT_60_tsangan_hhkb
LAYOUT_iso_regular_shift_regular_backspace -> LAYOUT_60_iso_tsangan
LAYOUT_iso_regular_shift_split_backspace -> LAYOUT_60_iso_tsangan_split_bs       (Not Community Layout)
LAYOUT_iso_split_shift_regular_backspace -> LAYOUT_60_iso_tsangan_split_rshift   (Not Community Layout)
LAYOUT_iso_split_shift_split_backspace -> LAYOUT_60_iso_tsangan_split_bs_rshift
     
dunk2k asked?
It can be seen there's also ABNT2/JIS split Right Shift (1u-1.75u) available on this board.
Would it be correct that matrix co-ordinates for 1u and 1.75u are 0,7 and 1,7 respectively?

You can view the standard layouts on the product page for the [F62](https://www.modelfkeyboards.com/product/f62-model-f-keyboard-kishsaver/).

The matrix co-ordinates when the full size right shift keycap is install:
The full size shift uses matrix location 1,7.

0,7 - There is no flipper installed. A  stabilizer insert for the shift key cap is installed.
1,7 - flipper installed.

The matrix co-ordinates for the spilt right shift:
0.7 - Smaller shift 
1,7 - 1u key. Typically Fn (MO1)

Reply:
> Can the '60_tsangan_hhkb" layout be changed to "60_ansi_tsangan_split_bs_rshift" to match the naming of the other tsangan community layouts.

I'm working on this amending this after my numerous Community Layout PRs have been merged. For now, `60_tsangan_hhkb` is what it's called so recommend to stick with it 😉

Reply:
My line of questioning regarding JIS/ABNT2 split Right Shift stems from [this](https://i.imgur.com/VU8anSf.jpeg).